x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/indexlifecycle/TimeSeriesLifecycleActionsIT.java

```diff
@@ -108,6 +109,18 @@ public void testDelete() throws Exception {
         assertBusy(() -> assertFalse(indexExists(index)));
     }
 
+    public void testDeleteOnlyShouldNotMakeIndexReadonly() throws Exception {
+        createIndexWithSettings(index, Settings.builder().put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 0));
+        createNewSingletonPolicy("delete", new DeleteAction(), TimeValue.timeValueHours(1));
+        updatePolicy(index, policy);
+        assertBusy(() -> {
+            assertThat(getStepKeyForIndex(index).getAction(), equalTo("complete"));
+            Map<String, Object> settings = getOnlyIndexSettings(index);
```

Comment:
Can you index a doc here to make sure it goes through okay? Checking the settings is good but it might be nice to be paranoid :)

Reply:
Done! I also added the same with an `expectThrows` to `testShrinkAction` and `testForceMergeAction` for completeness.